src/os_crypto/shared/keys.c

```diff
@@ -249,7 +249,14 @@ void OS_ReadKeys(keystore *keys, key_mode_t key_mode, int save_removed)
 
             *tmp_str = '\0';
             tmp_str++;
-            strncpy(id, valid_str, KEYSIZE - 1);
+            const int bytes_written = snprintf(id, sizeof(id), "%s", valid_str);
+
+            if ((size_t)bytes_written >= sizeof(id)) {
+                merror(INVALID_KEY, id);
+            }
+            else if (bytes_written < 0) {
+                merror("Error %d (%s).", errno, strerror(errno));
```

Comment:
If this will be a error message, what do you think if we append the error information to  the existing INVALID_KEY
```suggestion
                merror(INVALID_KEY " Error %d (%s).", id, errno, strerror(errno));
```